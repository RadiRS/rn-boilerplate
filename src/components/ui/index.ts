export { default as Text } from './text';
export { default as Button } from './button';
