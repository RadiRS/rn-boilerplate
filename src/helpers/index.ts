export * from './test-helpers/render-helper';
export * from './mask/mask-helpers';
export * from './utils';
