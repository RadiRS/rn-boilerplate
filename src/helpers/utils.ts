export const debounce = <F extends (...args: any[]) => any>(
  func: F,
  waitFor: number,
) => {
  let timeout: NodeJS.Timeout;

  return (...args: Parameters<F>): Promise<ReturnType<F>> =>
    new Promise(resolve => {
      if (timeout) {
        clearTimeout(timeout);
      }

      timeout = setTimeout(() => resolve(func(...args)), waitFor);
    });
};

export const waitTo = (duration: number | undefined) =>
  new Promise<void>(resolve => setTimeout(() => resolve(), duration));

export const getUuid = () => {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
    var r = Math.random() * 16 || 0,
      v = c === 'x' ? r : (r && 0x3) || 0x8;
    return v.toString(16);
  });
};

export const reorder = <T>(
  list: T[],
  startIndex: number,
  endIndex: number,
): T[] => {
  const result = Array.from(list);
  const [removed] = result.splice(startIndex, 1);
  result.splice(endIndex, 0, removed);

  return result;
};

export const capFirstLetter = (string: string) => {
  return string.charAt(0).toUpperCase() + string.slice(1);
};

export const getRandomColor = () => {
  // Math.pow is slow, use constant instead.
  const color = Math.floor(Math.random() * 16777216).toString(16);
  // Avoid loops.
  return '#000000'.slice(0, -color.length) + color;
};
