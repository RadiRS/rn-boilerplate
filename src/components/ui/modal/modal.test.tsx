import React from 'react';
import { render } from '@testing-library/react-native';

import { Modal, View } from '@/components/ui';
import { waitTo } from '@/helpers';

describe('<Modal />', () => {
  it('should render component correctly', async () => {
    const rendered = render(
      <Modal isVisible>
        <View />
      </Modal>,
    ).toJSON();

    // wait for modal animated to display
    await waitTo(500);

    expect(rendered).toMatchSnapshot();
  });
});
