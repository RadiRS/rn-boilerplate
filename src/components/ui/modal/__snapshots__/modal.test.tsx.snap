// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render component correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        [
          {
            "margin": 37.5,
            "transform": [
              {
                "translateY": 0,
              },
            ],
          },
          {
            "flex": 1,
            "justifyContent": "center",
          },
          [
            {
              "margin": 0,
            },
            {},
            undefined,
          ],
          {
            "margin": 0,
          },
        ],
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      collapsable={false}
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      style={
        {
          "flex": 1,
          "justifyContent": "center",
          "margin": 0,
          "transform": [
            {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View />
    </View>
  </View>
</Modal>
`;
